apps/web/tests/setup/global-mocks.ts

```
@@ -3,24 +3,184 @@
+      return createSupabaseMockResponse(data, null);
+    }),
+    maybeSingle: vi.fn().mockImplementation(() => createSupabaseMockResponse(null, null)),
+    csv: vi.fn().mockImplementation(() => createSupabaseMockResponse("", null)),
+    geojson: vi.fn().mockImplementation(() => createSupabaseMockResponse(null, null)),
+    explain: vi.fn().mockImplementation(() => createSupabaseMockResponse(null, null)),
+    rollback: vi.fn().mockImplementation(() => createSupabaseMockResponse(null, null)),
+    returns: vi.fn().mockReturnThis(),
+  };
+
+  // Add promise-like behavior for direct awaiting
+  (mockBuilder as any).then = function(onResolve: any, onReject?: any) {
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)